### ISSUE TITLE
False positive for used-before-assignement with conditional import

### ISSUE THREAD

**Issue by zozoens31** [PyCQA/pylint#3119]
### Steps to reproduce

Lint the following piece of code

```python3
import typing

if typing.TYPE_CHECKING:
    from typing_extensions import Literal

    AllowedValues = Literal['hello', 'world']
```

### Current behavior

pylint:E0601 Using variable 'Literal' before assignment (used-before-assignment)

### Expected behavior

The code should pass this rule (no errors)

### pylint --version output

pylint 2.4.0
astroid 2.2.5
Python 3.7.2 (default, Feb 12 2019, 08:15:36)
[Clang 10.0.0 (clang-1000.11.45.5)]
